File: ShellBoost.Samples/ShellBoost.Samples.Overview/OverviewShellFolderServer.cs
using ShellBoost.Core;

namespace ShellBoost.Samples.Overview
{
    public class OverviewShellFolderServer : ShellFolderServer
    {
        private SimpleFolder _root;

        protected override ShellFolder GetFolderAsRoot(ShellItemIdList idList)
        {
            if (_root == null)
            {
                _root = new SimpleFolder(idList);
            }
            return _root;
        }
    }
}

File: ShellBoost.Samples/ShellBoost.Samples.VBOverview/OverviewShellFolderServer.vb
Imports ShellBoost.Core

Namespace ShellBoost.Samples.Overview
    Public Class OverviewShellFolderServer
        Inherits ShellFolderServer

        Private _root As SimpleFolder

        Protected Overrides Function GetFolderAsRoot(ByVal idList As ShellItemIdList) As ShellFolder
            If _root Is Nothing Then
                _root = New SimpleFolder(idList)
            End If

            Return _root
        End Function
    End Class
End Namespace
